### DISCUSSION TITLE
SMG II default Wifi module

### DISCUSSION THREAD

**Discussion by Reggerriee** [syssi/esphome-smg-ii#48] [General]
Hello,

Is it possible to use this with the SMG II's own wifi module? If so, what does its configuration look like?

Thanks,
Rege

**Comment by syssi** [comment]
Could you open the OEM WiFi module? Is this dongle equipped with a ESP8266/ESP32? Please provide some high resolution photos if possible.

**Reply by syssi** [reply]
@Reggerriee Friendly reminder.

**Reply by Reggerriee** [reply]
Hello
sorry, I forgot.... tomorrow I will take pictures with the WIFI module :)

**Comment by Reggerriee** [comment]
![405505628_6981611191906380_197300559226033593_n](https://github.com/syssi/esphome-smg-ii/assets/5973632/b9d57e8c-2ce0-4da4-867b-d7426cf36092)
![377222062_747416223868794_3684505543221105825_n](https://github.com/syssi/esphome-smg-ii/assets/5973632/583c46b3-0c23-416b-a500-fca12e0c9d59)

**Reply by syssi** [reply]
This doesn't help. Could you remove the cover? I would like to see the chips on the PCB.

**Reply by Reggerriee** [reply]
I try :)

**Comment by Reggerriee** [comment]
![20231202_233514](https://github.com/syssi/esphome-smg-ii/assets/5973632/3ddb990e-975c-4fc9-b9c3-cb5c968d6094)
![20231202_233521](https://github.com/syssi/esphome-smg-ii/assets/5973632/0b4c8ebe-8449-4106-ae2b-56e7fc0c618a)
![20231202_233532](https://github.com/syssi/esphome-smg-ii/assets/5973632/41cef202-2964-4c8c-80ea-8653bdd5727d)
![20231202_233544](https://github.com/syssi/esphome-smg-ii/assets/5973632/9d052d8b-9367-40ab-bbb9-be9016801c26)
![20231202_233601](https://github.com/syssi/esphome-smg-ii/assets/5973632/8865a038-53e0-4b61-bbdb-1aba9b455958)

**Reply by S2000Stefan** [reply]
There is also a 3.0.0.1 version of the module.
Unfortunately, it only contains the EB-WF 03-01 chip and no ESP32.

![IMG_20231129_110049a](https://github.com/syssi/esphome-smg-ii/assets/35189051/5e36a534-4395-4219-a8ce-9cbbde92ad56)

**Reply by Reggerriee** [reply]
:'(
Do I then need to purchase the tools and build an ESP32 device, right? :)

**Reply by S2000Stefan** [reply]
I can't say. Let's wait and see what syssi has to say. He's the programmer. :)

**Reply by Reggerriee** [reply]
Okay thanks :)

**Reply by syssi** [reply]
I cannot find any details about the `EB-WF03-01`. I also strongly assume this isn't an ESP based microcontroller and you will have to build your ESP based device on your own (ESP32/ESP8266 dev board + MAX3232).

**Reply by Reggerriee** [reply]
I bought the device :)

This is okay?
![20231205_145604](https://github.com/syssi/esphome-smg-ii/assets/5973632/06ede236-1e20-4906-94c8-d9c3b503223e)
![20231205_145622](https://github.com/syssi/esphome-smg-ii/assets/5973632/a2f83f87-c073-44b9-8a53-3bd0279a2408)
![20231205_145631](https://github.com/syssi/esphome-smg-ii/assets/5973632/cf37756e-68cd-4c0f-9c7e-70d3a1f153b9)
![20231205_145649](https://github.com/syssi/esphome-smg-ii/assets/5973632/cfc21b73-ec0e-43cd-9075-31c7266388e7)

Another question:
Will this cable work? (I took a picture of both ends.)
![20231205_141831](https://github.com/syssi/esphome-smg-ii/assets/5973632/eaf705cd-fd59-4bcc-9a9d-f02d0c1a6e2c)
![20231205_141908](https://github.com/syssi/esphome-smg-ii/assets/5973632/3ec8ba44-0f91-458d-a2ae-ee9082f99888)

**Reply by syssi** [reply]
Please use a continuity tester to verify the correct pin out (on the d-sub 9pol side). I guess the connectors doesn't fit and two lines needs to be swapped.

**Comment by Reggerriee** [comment]
otherwise, why don't we make a Discord group where we can consult? :)

**Reply by syssi** [reply]
If you like feel free to contact me via discord. My handle is `syssi.` including the dot at the end.

**Comment by m0st1ted** [comment]
Hi. The original wifi dongle has ESP8266EX (EB-WF03-01 is ESP8266ex). I flashed it with esphome but it doesn't connect to HA api. It has low memory. I will change the chip with ESP32. Hope it works after.

**Reply by syssi** [reply]
Did you try to reduce the YAML a bit? Use `api` instead of `mqtt`. Reduce the log level etc.

**Reply by m0st1ted** [reply]
With the reduced YAML it works. I left only the sensors. The inverter is EASUN SMG III 6.2kW. I put 2 new "total_daily_energy" sensors for Mains power and Output power to be able to integrate it in "Energy Dashboard". It's pretty accurate.

**Reply by nheinemans** [reply]
Do you mind sharing how you managed to flash the original dongle?

**Reply by m0st1ted** [reply]
I remember that there is a reset pin witch must be connected to ground. In the end I used esp32. It has more ram. You can use all the sensors. It's more stable.

**Comment by Haso-M** [comment]
Did you the chip on original dongle?

**Reply by MBRetrofit** [reply]
I though that should be a good idea to replace the EB-WF03-01 with another ESP32/8266 on the WIFI Easun dongle. Any suggestion?

**Comment by MBRetrofit** [comment]
There is a new WIFI dongle for latest SMG II 6,2 KW. Has the same EB-WF03-01 and this is more likely a very chinese version of ESP8266 and is named "Tuya"

https://developer.tuya.com/en/docs/iot/wbr3-module-datasheet?id=K9dujs2k5nriy

![4](https://github.com/user-attachments/assets/78b9bfe0-5dfa-4994-9b3b-8ed9d792b80a)
![3](https://github.com/user-attachments/assets/3edc41ef-70da-40b2-b126-be13c3f756b2)
![2](https://github.com/user-attachments/assets/72933081-8c9d-4077-9abc-2e24cdb0a76b)
![1](https://github.com/user-attachments/assets/3ca1af06-af14-4764-95c2-d3f699c0e4a1)
